inspirehep/modules/multieditor/api.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Module for backend of multi record editor used in http://inspirehep.net."""
+
+from __future__ import absolute_import, print_function, division
+
+from flask import Blueprint, request, jsonify, session
+import copy
+
+from inspire_schemas.api import load_schema
+from inspirehep.modules.multieditor import tasks
+from inspirehep.modules.migrator.tasks import chunker
+from .actions import ActionFactory, compare_records
+from . import queries
+from .permissions import multieditor_use_permission
+
+
+match_type_map = {
+    'contains': 'contains',
+    'is equal to': 'exact',
+    'does not exist': 'missing',
+    'matches regular expression': 'regex'
+}
+
+blueprint = Blueprint(
+    'inspirehep_multieditor',
+    __name__,
+    template_folder='templates',
+    url_prefix='/multieditor',
+)
+
+
+@blueprint.route("/update", methods=['POST'])
+@multieditor_use_permission.require(http_exception=403)
+def update():
+    """Apply the user actions to the database records."""
+    user_actions = request.json['userActions']
+    all_selected = request.json['allSelected']
+    multieditor_session = session.get('multieditor_session', [])
+    if multieditor_session:
+        ids = multieditor_session['ids']
+        if all_selected:
+            ids = set(ids) - set(request.json['ids'])
+        else:
+            ids = set(ids) & set(request.json['ids'])
+    else:
+        return jsonify({'message': 'Please use the search before you apply actions'}), 400
+    actions = []
+    conditions = sanitize_user_conditions(user_actions.get('conditions', []))
+
+    for action in user_actions.get('actions', []):
+        actions.append(
+            ActionFactory.factory(action.get('actionName', ''),
+                                  **sanitize_user_action(action, conditions))
```

Comment:
It could be better to be consistent and move all the sanitization above:

```python
conditions = sanitize_user_conditions(user_actions.get('conditions', []))
actions = sanitize_user_actions(user_actions.get('actions', []))
```